sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -187,6 +187,16 @@ case class CreateTableAsSelect(
   }
 }
 
+/**
+ * Create a new table with the same table definition of the source table.
+ */
+case class CreateTableLike(
+    targetCatalog: TableCatalog,
+    targetTableName: Seq[String],
+    sourceCatalog: Option[TableCatalog],
+    sourceTableName: Seq[String],
```

Comment:
for the source table, what we really care is the table itself, not which catalog it comes from. I think it's better to define the plan as
```
case class CreateTableLike(
    targetCatalog: TableCatalog,
    targetTableName: Seq[String],
    sourceTable: NamedRelation,
    ifNotExists: Boolean)
```

In the planner, we match `CreateTableLike(..., r: DataSourceV2Relation, ..)`, and create the physical plan with source table `r.table`

Reply:
@cloud-fan I am a bit confused. The source can be both V1 and V2, right ? So how can we expect a DataSourceV2Relation as source ?

Reply:
the v1 table has a v2 adapter called `V1Table`. In `ResolveCatalogs`, we can lookup table from session catalog, create a `DataSourceV2Relation`, and pass it into `CreateTableLike`.